### DISCUSSION TITLE
Help with a Script: I can not limit MQTT sending frequency

### DISCUSSION THREAD

**Discussion by soulid** [arendst/Tasmota#15190] [Q&A] [answered]
### PROBLEM DESCRIPTION
Cannot limit MQTT sending frequency

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): ESP8266 with photodiode for reading out smart meter 
- [ ] Tasmota binary firmware version number used: 9.5
  - [X] Pre-compiled
  - [ ] Self-compiled
- [X] Flashing tools used: None
- [X] Provide the output of command: `Backlog Template; Module; GPIO 255`: 
```lua
  Configuration output here:
17:33:02.772 CMD: Backlog Template; Module; GPIO 255
17:33:02.859 MQT: stat/Stromzaehler/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}

```
- [X] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
17:34:28.454 CMD: Backlog Rule1; Rule2; Rule3
17:34:28.496 MQT: stat/Stromzaehler/RESULT = {"Command":"Unknown"}
17:34:28.699 MQT: stat/Stromzaehler/RESULT = {"Command":"Unknown"}
17:34:28.900 MQT: stat/Stromzaehler/RESULT = {"Command":"Unknown"}


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
17:35:06.958 CMD: Status 0
17:35:06.966 MQT: stat/Stromzaehler/STATUS = {"Status":{"Module":1,"DeviceName":"Stromzaehler","FriendlyName":["Stromzaehler"],"Topic":"Stromzaehler","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":0,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
17:35:06.975 MQT: stat/Stromzaehler/STATUS1 = {"StatusPRM":{"Baudrate":9600,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:22:56","StartupUTC":"2022-03-20T16:12:10","Sleep":50,"CfgHolder":4617,"BootCount":15,"BCResetTime":"2022-03-04T12:34:10","SaveCount":18,"SaveAddress":"FA000"}}
17:35:06.983 MQT: stat/Stromzaehler/STATUS2 = {"StatusFWR":{"Version":"9.5.0(tasmota)","BuildDateTime":"2021-11-08T00:56:32","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"371/699"}}
17:35:06.990 MQT: stat/Stromzaehler/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["WLS1924",""],"TelePeriod":10,"Resolution":"558180C0","SetOption":["00008009","2805C8000100060000005A0A000000000000","00000080","00006000","00000000"]}}
17:35:07.002 MQT: stat/Stromzaehler/STATUS4 = {"StatusMEM":{"ProgramSize":635,"Free":368,"Heap":18,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"14325E","FlashFrequency":40,"FlashMode":3,"Features":["00000809","87DAC787","043E8001","000000CF","010013C0","C000F989","00004004","00001000","00000020"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45","Sensors":"1,2,3,4,5,6,53"}}
17:35:07.011 MQT: stat/Stromzaehler/STATUS5 = {"StatusNET":{"Hostname":"Stromzaehler-4143","IPAddress":"192.168.0.150","Gateway":"192.168.0.1","Subnetmask":"255.255.255.0","DNSServer":"192.168.0.1","Mac":"C8:C9:A3:6A:B0:2F","Webserver":2,"WifiConfig":4,"WifiPower":17.0}}
17:35:07.018 MQT: stat/Stromzaehler/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.0.19","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_6AB02F","MqttUser":"openhabian","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
17:35:07.027 MQT: stat/Stromzaehler/STATUS7 = {"StatusTIM":{"UTC":"2022-03-20T16:35:07","Local":"2022-03-20T17:35:07","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"06:52","Sunset":"19:02"}}
17:35:07.037 MQT: stat/Stromzaehler/STATUS10 = {"StatusSNS":{"Time":"2022-03-20T17:35:07","MT175":{"E_in":5194.0,"E_out":7856.0,"P":-1077.00,"L1":-436.00,"L2":-38.00,"L3":-599.00,"Server_ID":"090149534b000485d342"}}}
17:35:07.045 MQT: stat/Stromzaehler/STATUS11 = {"StatusSTS":{"Time":"2022-03-20T17:35:07","Uptime":"0T00:22:57","UptimeSec":1377,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"WLS1924","BSSId":"DC:15:C8:5C:92:CC","Channel":11,"Mode":"11n","RSSI":46,"Signal":-77,"LinkCount":1,"Downtime":"0T00:00:04"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior: Continuous behavior

### EXPECTED BEHAVIOUR
Send MQTT data each 10 seconds instead of each 1 second

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._
![image](https://user-images.githubusercontent.com/179197/159172765-3baa4026-1050-4369-9898-4473a9c3b70e.png)

### ADDITIONAL CONTEXT
Script used to readout Iskra MT175:
>D
>B
->sensor53 r
>M 1
+1,3,s,16,9600,MT175
1,77070100010800ff@1000,E-gekauft,kWh,E_in,1
1,77070100020800ff@1000,E-geliefert,kWh,E_out,1
1,77070100100700ff@1,P,W,P,18
1,77070100240700ff@1,L1,W,L1,18
1,77070100380700ff@1,L2,W,L2,18
1,770701004C0700ff@1,L3,W,L3,18
1,77070100000009ff@#,Zaehlernummer,,Server_ID,0
#

Logging Period is well working 
17:38:30.433 MQT: tele/Stromzaehler/STATE = {"Time":"2022-03-20T17:38:30","Uptime":"0T00:26:20","UptimeSec":1580,"Heap":22,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"WLS1924","BSSId":"DC:15:C8:5C:92:CC","Channel":11,"Mode":"11n","RSSI":50,"Signal":-75,"LinkCount":1,"Downtime":"0T00:00:04"}}
17:38:30.443 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:38:30","MT175":{"E_in":5194.0,"E_out":7856.1,"P":-1098.00,"L1":-514.00,"L2":-11.00,"L3":-571.00,"Server_ID":"090149534b000485d342"}}

Inbetween above MQTT messages the following is send (**which is the issue**)
17:41:02.227 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:02","MT175":{"P":-930.00}}
17:41:02.247 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:02","MT175":{"L1":-463.00}}
17:41:02.268 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:02","MT175":{"L2":43.00}}
17:41:02.289 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:02","MT175":{"L3":-510.00}}
17:41:03.407 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:03","MT175":{"P":-933.00}}
17:41:03.429 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:03","MT175":{"L1":-461.00}}
17:41:03.449 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:03","MT175":{"L2":43.00}}
17:41:03.554 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:03","MT175":{"L3":-509.00}}
17:41:04.569 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:04","MT175":{"P":-930.00}}
17:41:04.590 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:04","MT175":{"L1":-460.00}}
17:41:04.611 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:04","MT175":{"L2":43.00}}
17:41:04.631 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:04","MT175":{"L3":-508.00}}
17:41:05.729 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:05","MT175":{"P":-930.00}}
17:41:05.750 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:05","MT175":{"L1":-457.00}}
17:41:05.771 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:05","MT175":{"L2":32.00}}
17:41:05.793 MQT: tele/Stromzaehler/SENSOR = {"Time":"2022-03-20T17:41:05","MT175":{"L3":-508.00}}

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by gemu2015** [comment, marked as answer]
check the docs!
your descriptor is defined to send every new value to MQTT.
1,77070100100700ff@1,P,W,P,18

the last entry is the number of digits + an optional flag to send the value as fast as possible code = 16
so 16+2 = 18 means 2 digits + fast send.
this will solve your issue:
 1,77070100100700ff@1,P,W,P,2

**Comment by soulid** [comment]
Hello gemu2015,
thank you for the helpful answer. I somehow read this part of the documentation before but couldn't interprete well. Now it is working as expected!
